imbi/clients/github.py

```diff
@@ -0,0 +1,290 @@
+                response.body,
+            )
+        if response.code == http.client.UNAUTHORIZED:
+            await self._refresh_token()
+            request_headers.update(self.headers)
+            response = await super().http_fetch(str(url),
+                                                method=method,
+                                                **kwargs)
+        return response
+
+    async def _refresh_token(self) -> None:
+        params = urllib.parse.urlencode({
+            'grant_type': 'refresh_token',
+            'refresh_token': self.token.refresh_token,
+            'client_id': self.token.integration.client_id,
+            'client_secret': self.token.integration.client_secret,
+        })
+        url = f'{self.token.integration.token_endpoint}?{params}'
+        self.logger.info('refreshing token for %s (%s)', self.user.username,
+                         self.token.external_id)
+        self.logger.debug('refresh_request: %r', params)
+        response = await super().http_fetch(
+            url,
+            method='POST',
+        )
+        if response.ok:
+            self.logger.debug('refresh response: %r', response.body)
+            await self.token.integration.upsert_user_tokens(
+                self.user.username,
+                self.token.external_id,
+                response.body['access_token'],
+                response.body['refresh_token'],
+                self.token.id_token,
+            )
+            self.token.access_token = response.body['access_token']
+            self.token.refresh_token = response.body['refresh_token']
+            self.headers['Authorization'] = f'Bearer {self.token.access_token}'
+        else:
+            self.logger.error(
+                'failed to refresh token for %s: %r',
+                self.user.username,
+                response.body,
+            )
+            raise GitHubAPIFailure(response, 'token refresh failed: %s',
+                                   response.code)
+
+    async def create_repository(self, org: str, name: str, project_id: int,
+                                namespace_slug: str,
+                                **attributes) -> Repository:
+        """Create a new repository on GitHub.
+
+        Args:
+            org: The organization to create the repository in
+            name: The name of the repository
+            project_id: The Imbi project ID for custom properties
+            namespace_slug: The namespace slug for custom properties
+            **attributes**: Additional repository attributes
+
+        Docs: https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-an-organization-repository
+        """  # noqa: E501
+        # Set default values for Imbi repositories
+        body = {
+            'name': name,
+            'has_issues': False,
+            'has_projects': False,
+            'has_wiki': False,
+            'auto_init': False,
+            'delete_branch_on_merge': True,
```

Comment:
What do you think about adding this to the configuration?
```yaml
automations:
  github:
    default_project_attributes:
      has_issues: false
      has_projects: false
      has_wiki: false
      delete_branch_on_merge: true
```

Reply:
I think that's a good idea